astropy/units/quantity_helper/function_helpers.py

```diff
@@ -1680,3 +1685,30 @@ def merge_arrays(
         unit,
         None,
     )
+
+
+@function_helper
+def average(a, axis=None, weights=None, returned=False, *, keepdims=np._NoValue):
+    # This override should no longer be needed once
+    # https://github.com/numpy/numpy/pull/30522 is merged (likely for
+    # "not NUMPY_LT_2_4_1").
+
+    a = _as_quantity(a)
+    a_value, a_unit = a.value, a.unit
+    weights = _as_quantity(weights)
```

Comment:
This raises an error for `weights=None`, no?

Hmm, just checked, that indeed errors. There should have been a test for that... Let me push the fix directly to your branch.